src/test/java/io/jenkins/plugins/analysis/warnings/ReferenceFinderITest.java

```diff
@@ -0,0 +1,740 @@
+package io.jenkins.plugins.analysis.warnings;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Consumer;
+
+import org.junit.Test;
+
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+
+import io.jenkins.plugins.analysis.core.history.ReferenceFinder;
+import io.jenkins.plugins.analysis.core.model.AnalysisResult;
+import static io.jenkins.plugins.analysis.core.model.Assertions.*;
+import io.jenkins.plugins.analysis.core.steps.IssuesRecorder;
+import io.jenkins.plugins.analysis.core.steps.ToolConfiguration;
+import io.jenkins.plugins.analysis.core.testutil.IntegrationTest;
+import io.jenkins.plugins.analysis.core.views.ResultAction;
+
+import hudson.model.FreeStyleBuild;
+import hudson.model.FreeStyleProject;
+import hudson.model.Result;
+import hudson.tasks.Shell;
+
+/**
+ * Integration tests of the warnings plug-in in freestyle jobs. Tests the new reference finder {@link ReferenceFinder}.
+ *
+ * @author Arne Sch�ntag
+ */
+public class ReferenceFinderITest extends IntegrationTest {
+
+    private String jobName = "Job";
+    private String refName = "RefJob";
+
+    /**
+     * Checks if the reference is taken from the last successful build and therefore returns a success in the end.
+     */
+    @Test
+    public void shouldCreateSuccessResultWithIgnoredUnstableInBetween() throws IOException, InterruptedException {
+        FreeStyleProject project = createJobWithWorkspaceFile(jobName, "eclipse2Warnings.txt");
+        enableWarnings(project, publisher -> {
+            publisher.setUnstableNewAll(3);
+        });
+
+        AnalysisResult result;
+        result = scheduleBuildAndAssertStatus(project, Result.SUCCESS);
+        assertThat(result).hasTotalSize(2);
+        assertThat(result).hasOverallResult(Result.SUCCESS);
+
+        project.getSomeWorkspace().deleteContents();
+        copyFilesToWorkspace(project, "eclipse.txt");
```

Comment:
Beide Zeilen auslagern in Methode, z.B. replaceWorkspaceWith(project, fileName) und überall ersetzen lassen (Extract Method Refactoring)